src/Middleware/OutputCaching/src/CacheVaryByRules.cs

```diff
@@ -11,14 +11,14 @@ namespace Microsoft.AspNetCore.OutputCaching;
 /// </summary>
 public sealed class CacheVaryByRules
 {
-    private Dictionary<string, string>? _varyByCustom;
+    private Dictionary<string, string>? _varyByValues;
 
-    internal bool HasVaryByCustom => _varyByCustom != null && _varyByCustom.Any();
+    internal bool HasVaryByValues => _varyByValues != null && _varyByValues.Any();
 
     /// <summary>
-    /// Gets a dictionary of key-pair values to vary the cache by.
+    /// Gets a dictionary of key-pair values to vary by.
     /// </summary>
-    public IDictionary<string, string> VaryByCustom => _varyByCustom ??= new();
+    public Dictionary<string, string> VaryByValues => _varyByValues ??= new ();
```

Comment:
```suggestion
    public IDictionary<string, string> VaryByValues => _varyByValues ??= new ();
```

Reply:
Any reason we should expose the interface instead? it's `get` only. Just in case we need to change the actual internal implementation?

Reply:
1. This didn't go through API review
2. We generally don't expose concrete collection types in public API